### PR TITLE
libavcodec/dxva2_av1.c: fixed superres_denom param

### PR DESCRIPTION
**Pull request by tong1wu**
fixed superres_denom to solve the problem of the denominator being 0.

### CODE REVIEW

**Comment by tong1wu** [comment]
@xuguangxin @xhaihao @feiwan1 please help me review this patch and the last patch, thanks.

File: libavcodec/dxva2_av1.c [hunk 1]
```diff
@@ -73,7 +73,7 @@ static int fill_picture_parameters(const AVCodecContext *avctx, AVDXVAContext *c
     pp->max_height = seq->max_frame_height_minus_1 + 1;
 
     pp->CurrPicTextureIndex = ff_dxva2_get_surface_index(avctx, ctx, h->cur_frame.tf.f);
-    pp->superres_denom      = frame_header->use_superres ? frame_header->coded_denom : AV1_SUPERRES_NUM;
+    pp->superres_denom      = frame_header->use_superres ? frame_header->coded_denom + AV1_SUPERRES_DENOM_MIN : AV1_SUPERRES_NUM;
```
Comment (xuguangxin) [inline libavcodec/dxva2_av1.c:76]: why we need this change, could you provide some comments in the commit log? the comment better link to some reference.

**Comment by feiwan1** [comment]
Better to change title to "avcodec/dxva2: XXXXX".

File: libavcodec/dxva2_av1.c (hunk 1 above)
Comment (xuguangxin) [reply libavcodec/dxva2_av1.c:76]: Defined in spec 5.9.8. When superres is enabled, SuperresDenom equals "coded_denom + SUPERRES_DENOM_MIN" not  coded_denom. 
 
Maybe this is better.

Comment (tong1wu) [reply libavcodec/dxva2_av1.c:76]: ok, just modified it like that

**Comment by xuguangxin** [comment]
@tong1wu , just for double-check. It's not used by DXVA2 only, D3D11 will use the code too, right?

**Comment by tong1wu** [comment]
@xuguangxin yes basically dxva2 and d3d11 both use the dxva2_av1.c to fill parameters.

**Comment by xhaihao** [comment]
upstream patch: http://ffmpeg.org/pipermail/ffmpeg-devel/2021-September/284920.html

**Pull request closed without merging**